### DISCUSSION TITLE
Using trivy from go

### DISCUSSION THREAD

**Discussion by pujan14** [aquasecurity/trivy#5823] [Q&A] [answered]
I am trying to use trivy from go programmatically, Is this supported? Are there any examples?
I'm trying to scan specific path on file system and generate SBOM from that, so pointer will be highly appreciated.

**Comment by nikpivkin** [comment, marked as answer]
Hi @pujan14 !

The Trivy code is in the `pkg` package, so it is free to be imported into your code. You can find examples of how to use trivy in publicly available [repositories](https://github.com/search?q=github.com%2Faquasecurity%2Ftrivy+path%3A*%2Fgo.mod&type=code).
